packages/appsync-modelgen-plugin/src/visitors/appsync-java-visitor.ts

```diff
@@ -33,15 +36,22 @@ export class AppSyncModelJavaVisitor<
   protected additionalPackages: Set<string> = new Set();
   protected usingAuth: boolean = false;
 
+  // We override lazy generation to also only generate lazy models if data store is disabled
+  protected override isGenerateModelsForLazyLoadAndCustomSelectionSet(): boolean {
+    return super.isGenerateModelsForLazyLoadAndCustomSelectionSet() && !(this.config.isDataStoreEnabled ?? false);
+  }
+
   generate(): string {
     // TODO: Remove us, leaving in to be explicit on why this flag is here.
     const shouldUseModelNameFieldInHasManyAndBelongsTo = true;
     // This flag is going to be used to tight-trigger on JS implementations only.
     const shouldImputeKeyForUniDirectionalHasMany = false;
     this.processDirectives(shouldUseModelNameFieldInHasManyAndBelongsTo, shouldImputeKeyForUniDirectionalHasMany);
 
-    if (this._parsedConfig.generate === 'loader') {
+    if (this._parsedConfig.generate === CodeGenGenerateEnum.loader) {
       return this.generateClassLoader();
+    } else if (this._parsedConfig.generate === CodeGenGenerateEnum.metadata) {
+      return this.generateModelPathClasses();
     }
```

Comment:
I was curious when are these used? It looks like the tests are using `CodeGenGenerateEnum.code` so I'm assuming that's the default for `amplify codegen models` from an Android project. How does the developer choose to generate "loader" or "metadata" over "code"?

Reply:
It comes from preset.ts `generate: 'metadata'`

Reply:
Oh i see it now, thanks for pointing that out